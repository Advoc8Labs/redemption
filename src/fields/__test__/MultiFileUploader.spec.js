import React from 'react';
import { create, act } from 'react-test-renderer';
import { MultiFileUploader } from '..';

describe('MultiFileUploader', () => {
    it('renders a basic mutli-file upload component', () => {
        // Arrange
        const props = {
            field: {
                field_height: '50',
                field_height_unit: '%',
                field_width: '50',
                field_width_unit: 'px',
                background_color: '000000',
                servar: {
                    key: 'key',
                    name: 'Upload file...',
                    required: true,
                    metadata: {
                        file_types: 'image/*'
                    }
                },
                metadata: {
                    icon_url: 'https://feathery.tech',
                    uploader_padding_top: '0',
                    uploader_padding_bottom: '0',
                    uploader_padding_left: '0',
                    uploader_padding_right: '0',
                    cta_padding_top: '0',
                    cta_padding_bottom: '0',
                    cta_padding_left: '0',
                    cta_padding_right: '0'
                }
            },
            onChange: jest.fn(),
            onClick: jest.fn()
        };

        // Act
        let component;
        act(() => {
            component = create(<MultiFileUploader {...props} />);
        });

        // Assert
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
