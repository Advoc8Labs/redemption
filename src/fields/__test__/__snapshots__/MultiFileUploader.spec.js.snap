// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiFileUploader renders a basic mutli-file upload component 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexWrap": "wrap",
    }
  }
>
  <div
    id="key"
    onClick={[Function]}
    style={
      Object {
        "alignItems": "center",
        "border": "1px solid lightgrey",
        "borderRadius": "4px",
        "cursor": "pointer",
        "display": "flex",
        "flexDirection": "column",
        "height": "50%",
        "justifyContent": "space-between",
        "maxHeight": "100%",
        "overflow": "hidden",
        "paddingBottom": "0px",
        "paddingLeft": "0px",
        "paddingRight": "0px",
        "paddingTop": "0px",
        "position": "relative",
        "width": "50px",
      }
    }
  >
    <img
      className="img-fluid"
      src="https://feathery.tech"
      style={
        Object {
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": 0,
        }
      }
    />
    <div
      style={
        Object {
          "background": "#000000",
          "paddingBottom": "0px",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "paddingTop": "0px",
        }
      }
    >
      Upload file...
    </div>
    <input
      accept="image/*"
      multiple={true}
      onChange={[Function]}
      required={true}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
  </div>
</div>
`;
